core/primitives/src/sharding.rs

```diff
@@ -194,9 +195,29 @@ impl ShardChunkHeaderV3 {
         tx_root: CryptoHash,
         prev_validator_proposals: Vec<ValidatorStake>,
         congestion_info: Option<CongestionInfo>,
+        permanent_contracts_metadata: Vec<(AccountId, CryptoHash)>,
         signer: &ValidatorSigner,
     ) -> Self {
-        let inner = if let Some(congestion_info) = congestion_info {
+        let inner = if ProtocolFeature::GlobalContracts.enabled(protocol_version) {
+            let congestion_info = congestion_info.expect("Congestion info is required");
+            ShardChunkHeaderInner::V4(ShardChunkHeaderInnerV4 {
+                prev_block_hash,
+                prev_state_root,
+                prev_outcome_root,
+                encoded_merkle_root,
+                encoded_length,
+                height_created: height,
+                shard_id,
+                prev_gas_used,
+                gas_limit,
+                prev_balance_burnt,
+                prev_outgoing_receipts_root,
+                tx_root,
+                prev_validator_proposals,
+                congestion_info,
+                new_permanent_contracts: permanent_contracts_metadata,
```

Comment:
nit: keep the naming consistent

Reply:
actually can we name it global_contracts_metadata? (permanent --> global)